### ISSUE TITLE
Can not find the reason why the following code doesn't work.

### ISSUE THREAD

**Issue by lymanzhang** [wblut/HE_Mesh#57]
Hi Frederik, 

I just create a mesh based on polygons and transfer the data to lattice modification function, but lattice modification can not work. I tweak the example Ref_HEC_FromPolygons to get the following code. And the  He_Mesh lib I just used is version_5.0.2-20170205.

```
import wblut.math.*;
import wblut.processing.*;
import wblut.core.*;
import wblut.hemesh.*;
import wblut.geom.*;


HE_Mesh mesh;
WB_Render render;


void setup() {
  size(1000, 1000, P3D);
  
  //create base points for a single hexagon
  WB_Point[] basepoints =new WB_Point[6];
  for (int i=0;i<6;i++) {
   basepoints[i]=new WB_Point(0,100,0);
   if(i>0) basepoints[i].rotateAboutAxisSelf(Math.PI/3.0*i,0,0,0,0,0,1);
  }
  
  //create polygons from shifted copies of base points
  WB_Polygon[] polygons=new WB_Polygon[7];
  double ds=Math.cos(Math.PI/6)*200;
  polygons[0]=new WB_Polygon(basepoints);
  for(int i=0;i<6;i++){
    WB_Point[] localpoints=new WB_Point[6];
    for(int j=0;j<6;j++){
     localpoints[j]=basepoints[j].add(ds,0,0);//Shift base polygon
     localpoints[j].rotateAboutAxisSelf(Math.PI/3.0*i,0,0,0,0,0,1);//Rotate shifted polygon
    }
    polygons[i+1]=new WB_Polygon(localpoints);
  }
  
  HEC_FromPolygons creator=new HEC_FromPolygons();
  
  creator.setPolygons(polygons);
  //alternatively polygons can be any Collection<WB_Polygon>
  mesh=new HE_Mesh(creator); 
 
 
  HET_Diagnosis.validate(mesh);
  
  // lattice modify
  HEM_Lattice modifierLattice=new HEM_Lattice();
  
  modifierLattice.setWidth(2);// desired width of struts
  modifierLattice.setDepth(2);// depth of struts
  modifierLattice.setThresholdAngle(1.5*HALF_PI);// treat edges sharper than this angle as hard edges
  modifierLattice.setFuse(true);// try to fuse planar adjacent planar faces created by the extrude
  modifierLattice.setFuseAngle(0.1*HALF_PI);// threshold angle to be considered coplanar
  
  mesh.modify(modifierLattice);
  // lattice modify
  
  render=new WB_Render(this);
}

void draw() {
  background(55);
  directionalLight(255, 255, 255, 1, 1, -1);
  directionalLight(127, 127, 127, -1, -1, 1);
  translate(width/2,height/2);
  rotateY(mouseX*1.0f/width*TWO_PI);
  rotateX(mouseY*1.0f/height*TWO_PI);
  stroke(0);
  render.drawEdges(mesh);
  noStroke();
  render.drawFaces(mesh);
}
```

**Comment by wblut** [comment]
HEM_Lattice appears to be broken for open surfaces.

**Comment by wblut** [comment]
Fixed in 5.0.3

**Comment by lymanzhang** [comment]
it works well now, thanks!
